src/components/Identity/index.js

```diff
@@ -74,10 +78,18 @@ const createIdentity = ({ config, logger, cookieJar }) => {
       },
       // Waiting for opt-in because we'll be reading the ECID from a cookie
       // TO-DOCUMENT: We wait for ECID before trigger any events.
-      onBeforeDataCollection(payload) {
+      onBeforeDataCollection(payload, processedIds) {
```

Comment:
It feels incorrect to pass the ids into this hook, since this hook is app wide; Hooks' signatures are predefined so that components know how to implement them.